modules/qdldl/0.1.7/patches/add_build_file.patch

```diff
@@ -0,0 +1,71 @@
+diff --git BUILD.bazel BUILD.bazel
+new file mode 100644
+index 0000000..12271b7
+--- /dev/null
++++ BUILD.bazel
+@@ -0,0 +1,65 @@
++load("@rules_cc//cc:defs.bzl", "cc_library")
++
++load(
++    "//bazel:cmake_configure_file.bzl",
++    "cmake_configure_file",
++)
++
++licenses(["notice"])
++
++package(
++    default_visibility = ["//visibility:public"],
++)
++
++cmake_configure_file(
++    name = "configure_file_types",
++    src = "configure/qdldl_types.h.in",
++    out = "include/qdldl_types.h",
++    # https://github.com/osqp/qdldl/blob/v0.1.6/README.md#custom-types-for-integer-floats-and-booleans
++    defines = [
++        "QDLDL_BOOL_TYPE=unsigned char",
++        # Keep the `int` type sync'd with the build of @osqp_internal and
++        # @scs_internal. See drake/tools/workspace/qdldl_internal/README.md.
++        "QDLDL_INT_TYPE=int",
++        "QDLDL_LONG=0",
++        # Keep the `int` type sync'd with the build of @osqp_internal and
++        # @scs_internal. See drake/tools/workspace/qdldl_internal/README.md.
++        "QDLDL_FLOAT_TYPE=double",
++        "QDLDL_FLOAT=0",
++        # Match this to QDLDL_INT_TYPE above.
++        "QDLDL_INT_TYPE_MAX=INT_MAX",
++    ],
++    visibility = ["//visibility:private"],
++)
++
++cmake_configure_file(
++    name = "configure_file_version",
++    src = "configure/qdldl_version.h.in",
++    out = "include/qdldl_version.h",
++    cmakelists = [
++        "CMakeLists.txt",
++    ],
++    visibility = ["//visibility:private"],
++)
++
++cc_library(
++    name = "qdldl",
++    hdrs = [
++        "include/qdldl.h",
++        ":include/qdldl_types.h",
++        ":include/qdldl_version.h",
++    ],
++    srcs = [
++        "src/qdldl.c",
++    ],
++    copts = [
++        "-fvisibility=hidden",
++    ] + select({
++        "@platforms//os:windows": [],
++        "//conditions:default": ["-Werror=incompatible-pointer-types"],
++    }),
++    includes = [
++        "include",
++    ],
++    linkstatic = 1,
```

Comment:
Is this necessary? It does restrict users.

Reply:
remove at https://github.com/bazelbuild/bazel-central-registry/pull/1421/commits/4d48b5640e78bce45a082232f37ea787b692f8b0